src/Compilers/CSharp/Portable/Binder/UsingStatementBinder.cs

```diff
@@ -82,7 +81,28 @@ internal override BoundStatement BindUsingStatementParts(DiagnosticBag diagnosti
                 if (!iDisposableConversion.IsImplicit)
                 {
                     TypeSymbol expressionType = expressionOpt.Type;
-                    if ((object)expressionType == null || !expressionType.IsErrorType())
+                    MethodSymbol disposeMethod = expressionType == null ? null : SatisfiesDisposePattern(expressionOpt.Type, diagnostics);
+                    if ((object)disposeMethod != null)
+                    {
+                        if (!disposeMethod.ReturnsVoid)
```

Comment:
This seems like a check for `SatisfiesDisposePattern`. If the signature didn't match, how did it satisfy the check?